<a href="https://colab.research.google.com/github/jordannemirovsky/DS1002/blob/main/Mar25HW.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [46]:
import pandas as pd

In [47]:
ride_sharing = pd.read_csv('./ride_sharing.csv')

In [48]:
ride_sharing.shape

(25760, 10)

There are 25760 rows and 10 columns.

In [49]:
ride_sharing['duration'] = ride_sharing['duration'].str.extract('(\d+)').astype(int)

In [50]:
print(ride_sharing.dtypes)

duration            int64
station_A_id        int64
station_A_name     object
station_B_id        int64
station_B_name     object
bike_id             int64
user_type           int64
user_birth_year     int64
user_gender        object
tire_size           int64
dtype: object


In [51]:
ride_sharing['duration'].min()

1

The min ride duration is 1 minute.

In [52]:
ride_sharing['duration'].max()

1372

The max ride duration is 1372 minutes.

In [53]:
ride_sharing_new = ride_sharing.query('duration <= 60')

In [54]:
ride_sharing_new['duration'].min()

1

The min ride duration is 1 minute.

In [55]:
ride_sharing_new['duration'].max()

60

The max ride duration is 60 minutes.

In [56]:
ride_sharing_new = ride_sharing_new.query('tire_size == 26')

In [57]:
count = len(ride_sharing_new.query('user_gender == "Female" and duration <= 60'))
count

2962

2962 records are female riders with 26-inch wheeled bikes who rent for one hour or less.

In [79]:
more_messy_data = pd.read_csv('./more-messy-data.csv')

In [80]:
print(more_messy_data.duplicated().sum())

287


There are 287 duplicate rows.

In [81]:
more_messy_data = more_messy_data.drop_duplicates()

In [82]:
columns = ['sepal_length', 'sepal_width', 'petal_length', 'petal_width']
more_messy_data[columns] = more_messy_data[columns].fillna(more_messy_data[columns].mean())

<ipython-input-82-c9339e9f5a73>:2: FutureWarning: The default value of numeric_only in DataFrame.mean is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.
  more_messy_data[columns] = more_messy_data[columns].fillna(more_messy_data[columns].mean())


In [83]:
more_messy_data = more_messy_data.dropna()

In [84]:
species = more_messy_data['species'].nunique()
species

9

There are 9 species.

In [85]:
#finding the names of the 9 species
nine_species = more_messy_data['species'].unique()
print(nine_species)

['virginica' 'setosa' 'Versicolor' 'setoso' 'Viginica' 'Virginia' 'Setosa'
 'Setosan' 'versicolor']


In [86]:
#creating a dictionary
mapping = {
    'setosa': 'Setosa',
    'setoso': 'Setosa',
    'Setosa': 'Setosa',
    'Setosan': 'Setosa',
    'Versicolor': 'Versicolor',
    'versicolor': 'Versicolor',
    'virginica': 'Virginica',
    'Viginica': 'Virginica',
    'Virginia': 'Virginica'
}

#making three different values for the species variable
more_messy_data['species'] = more_messy_data['species'].replace(mapping)

In [87]:
#checking to see there are 3 species
species_check = more_messy_data['species'].nunique()
species_check

3

In [88]:
#checking to see what the 3 species are
three_species = more_messy_data['species'].unique()
print(three_species)

['Virginica' 'Setosa' 'Versicolor']
